technical-reference/reference-technical-articles/technical-doc-token-converter.md

```diff
@@ -0,0 +1,226 @@
+
+The following diagram illustrates the flow of funds between contracts and the architecture it has:
+
+<figure><img src="../../.gitbook/assets/token_converter_funds.svg" alt="Integration of Token converters in the Venus protocol"><figcaption>Integration of Token converters in the Venus protocol</figcaption></figure>
+
+- The transfer of funds to and from the `ProtocolShareReserve` contract are initiated by external agents (permissionless).
+- The transfers of funds from the Token converters to their destinations are performed on each conversion. The conversions are performed by external agents (permissionless).
+- The transfer of XVS from the `XVSVaultTreasury` to the `XVSVault` will be performed via VIP.
+
+`RiskFundConverter` and `SingleTokenConverters` possess the ability to "facilitate" token conversions for the income they receive, obtaining the desired tokens (base asset) by relying on oracle prices as reference points to accept conversions that external agents will carry out.
+
+Venus encourages these conversions by providing incentives that create arbitrage opportunities in the market, with the expectation that external agents will seize upon these opportunities.
+
+The diagram below illustrates the connection between income harvesting (the process by which Venus transfers income from its sources to the `ProtocolShareReserve`), distribution (application of the protocol's Tokenomics-defined rules), and conversion (acquiring the required base assets).
+
+<figure><img src="../../.gitbook/assets/token_converters_sequence.svg" alt="Diagram showing the relationship among the harvesting, distribution, and converting of income"><figcaption>Relationship among the harvesting, distribution, and converting of income</figcaption></figure>
+
+Regarding the destinations of the Token converters:
+
+- `RiskFund` already includes the integration with the `Shortfall` contract. The USDT received in the conversions will be [auctioned off in the Shortfall contract](shortfall-and-auctions.md).
+- `XVSVaultTreasury` accumulates the XVS received in the conversions; those XVS will be sent to the `XVSVault` eventually via VIP, which will update also the APR in the `XVSVault`
+- `PrimeLiquidityProvider` accumulates [the Prime funds](prime.md), distributing them according to the speeds configured via VIP.
+
+# `AbstractTokenConverter`
+
+`AbstractTokenConverter` contract provides every feature to configure and convert the desired assets. The `RiskFundConverter` and the `SingleTokenConverters` extend this abstract contract (that could also be extended by any other contract in the future because it provides their features in an agnostic way).
```

Comment:
```suggestion
`AbstractTokenConverter` contract provides required features to configure and convert desired assets. The `RiskFundConverter` and the `SingleTokenConverters` extend this abstract contract (that could also be extended by other contracts in the future because it provides these features agnostically).
```